src/main/java/frc/team2412/robot/util/TimeBasedMedianFilter.java

```diff
@@ -0,0 +1,65 @@
+package frc.team2412.robot.util;
+
+import java.util.ArrayDeque;
+import java.util.Queue;
+
+import edu.wpi.first.wpilibj.Timer;
+
+/**
+ * A class that implements a median filter based on how recent information is.
+ */
+public class TimeBasedMedianFilter {
+
+    private static class TimeData {
+        public final double time;
+        public final double data;
+
+        public TimeData(double time, double data) {
+            this.time = time;
+            this.data = data;
+        }
+    }
+
+    private final Queue<TimeData> values;
+    private final double filterTime;
+
+    /**
+     * Creates a new {@code TimeBasedMedianFilter}.
+     *
+     * @param filterTime
+     *            Length of time in seconds that inputs affect average.
+     */
+    public TimeBasedMedianFilter(double filterTime) {
```

Comment:
Why not use the built in wpilib median filter with a number that matches the duration in the median?

Reply:
The reason we chose not to use [`MedianFilter`](https://first.wpi.edu/wpilib/allwpilib/docs/release/java/edu/wpi/first/math/filter/MedianFilter.html) is because we expect the distance to change smoothly over time, not necessarily the different samples. (For example, if we pause data collection- Maybe because we're doing manual controls- we wouldn't expect the distance to be smooth, since we might have moved in the meantime.)
We could have solved this by polling NetworkTables solely in periodic and updating a variable that we return in `getDistance()`, but I personally didn't like unnecessarily getting data. We could also have maybe tried to use [`MedianFilter`](https://first.wpi.edu/wpilib/allwpilib/docs/release/java/edu/wpi/first/math/filter/MedianFilter.html) in a way that was based off of time, but I couldn't think of anything that wasn't essentially reimplementing it.